### PR TITLE
Fix internally calling other verta services apis logic

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java [hunk 1]
```diff
@@ -88,7 +87,7 @@ public void run() {
         LOGGER.error("DeleteEntitiesCron Exception: ", ex);
       }
     }
-    isBackgroundUtilsCall = false;
+    ModelDBUtils.unregisteredBackgroundUtilsCount();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java]: this should be in a finally ?

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java [hunk 2]
```diff
@@ -708,7 +706,7 @@ private static void runMigration() {
                   LOGGER.warn(
                       "ModelDBHibernateUtil runMigration() getting error : {}", e.getMessage(), e);
                 }
-                isBackgroundUtilsCall = false;
+                ModelDBUtils.unregisteredBackgroundUtilsCount();
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java]: same

File: backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/cron_jobs/DeleteEntitiesCron.java]: updated

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBHibernateUtil.java]: updated

**Review by ravishetye — APPROVED** [review]

**Pull request merged**